### PR TITLE
New JSON Parser

### LINKED ISSUE #9197: Create a new JSON Parser
Goal: Create a new and more flexible JSON Parser.

### PR DESCRIPTION
**Pull request by sspaink**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

Close #9197, #1363

This pull request introduces a new JSON Parser called `json_v2` that is intended to be more flexible then the existing JSON parser.

The query language being used will still be [GJSON Path Syntax](https://github.com/tidwall/gjson/blob/v1.7.5/SYNTAX.md), but the way to configure this parser has changed significantly. I've provided multiple configuration examples within the `testdata` directory under the `json_v2 directory`. The README.md also goes into more detail on how the configuration works.

Issues closed:

- Allow setting tags inside or outside of a nested array 
    - Closes: #6437 (The testdata directory named "nested_and_nonnested_tags" uses the JSON from this issue)
    - Closes: #6853 (The testdata directory named "nested_tags" and "nested_tags_complex" use the JSON from this issue)
- Fields in the JSON can be of any type (int,float,string,bool)
    - Closes: #7097  (The testdata directory named "types" covers this issue)
- Set measurement name independently from other queries and allow any type 
    - Closes:  #6990 (The testdata directory named "nested_array_of_objects" uses the JSON from this issue)
    - Closes: #7212 (The testdata directory named "measurement_name_int" covers this issue
- Set timestamp independently from other queries
    - Closes: #5940 (the testdata directory named "timestamp" covers this issue)

### CODE REVIEW

**Comment by samhld** [comment]
@sspaink @sjwang90 

This is so cool!  I tested it out and it works great!  I'll file the "zip" feature request today.

Two notes from testing:
* I'm wondering if the syntax of `[[xxx.xxx.json_v2.object]]` and `[[xxx.xxx.json_v2.tag]]`, etc. is unnecessary typing?  In the examples, the `inputs.file` portion seems to be implied, right?  Is there a technical reason we wouldn't revert back to simply `[[object]]`, `[[tag]]`, etc.?  The TOML indentations seem to make it clear what the prefixes would be IMO.
* If we do stick with the longer form, it might be good to explain what the `xxx` notation is in the example configurations (maybe just a comment `// where 'xxx' is ...`)

**Comment by sspaink** [comment]
@samhld thanks for the feedback! This is a limitation of TOML that sub-tables needs the parent name. I agree it would be nicer if it was just object/field/tag. I did update the README to be more clear and removed "xxx" with an actual example.

**Comment by samhld** [comment]
@sspaink Oh gotcha!  TIL!

File: plugins/parsers/json_v2/README.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+* If a type is explicitly defined, the parser will enforce this type and convert the data to the defined type if possible. If the type can't be converted then the parser will fail.
+* If a type isn't defined, the parser will use the default type defined in the JSON (int, float, string)
+
+The type values you can set:
+
+* `int`, bool, floats or strings (with valid numbers) can be converted to a int.
+* `uint`, bool, floats or strings (with valid numbers) can be converted to a uint.
+* `string`, any data can be formatted as a string.
+* `float`, string values (with valid numbers) or integers can be converted to a float.
+* `bool`, the string values "true" or "false" (regardless of capitalization) or the integer values `0` or `1`  can be turned to a bool.
```
[the remark this answers is not in the document]
Comment (sspaink) [reply plugins/parsers/json_v2/README.md:187]: Good idea, we should look into adding this. Need to get some example configs from v1.

**Review by reimda — APPROVED** [review]
Good job!

**Pull request merged**